Cython/Compiler/Nodes.py

```diff
@@ -5310,6 +5307,16 @@ def analyse_declarations(self, env):
         self.scope = scope = self.entry.type.scope
         if scope is not None:
             scope.directives = env.directives
+            if "dataclasses.dataclass" in env.directives:
+                is_frozen = False
+                dataclass_directives = env.directives["dataclasses.dataclass"]
+                if dataclass_directives:
+                    frozen_directive = dataclass_directives[1].get('frozen', None)
```

Comment:
Hmm, whatever `dataclass_directives[1]` is – probably some kind of dict, but reading code like this suggests that there might be a more descriptive way to represent these "directives".

Note that you don't need to spell out the `None` fallback value for `dict.get()`.
```suggestion
                    frozen_directive = dataclass_directives[1].get('frozen')
```

Reply:
(I was referring to the `dataclass_directives` … tuple, maybe? Saying `something[1]` simply doesn't give much information about what we're getting as a result. And I don't mean so much the type of the result as the semantic meaning. Attributes are much clearer since they give a name to the thing that we're looking at. That's why named tuples are better than regular tuples in some cases. Well, and dataclasses.)